src/api/PerpV2LeverageViewerAPI.ts

```diff
@@ -0,0 +1,157 @@
+   * @return                          The address of the ERC20 collateral token
+   */
+   public async getCollateralTokenAsync(
+    callerAddress: Address = undefined,
+  ): Promise<Address> {
+    return await this.perpV2LeverageModuleViewerWrapper.collateralToken(callerAddress);
+  }
+
+  /**
+   * Returns the maximum amount of Sets that can be issued. Because upon issuance we lever up the Set
+   * before depositing collateral there is a ceiling on the amount of Sets that can be issued before the max
+   * leverage ratio is met. In order to accurately predict this amount the user must pass in an expected
+   * slippage amount, this amount should be calculated relative to Index price(s) of vAssets held by the Set,
+   * not the mid-market prices. The formulas used here are based on the "conservative" definition of free
+   * collateral as defined in PerpV2's docs.
+   *
+   * @param setTokenAddress           Instance of SetToken
+   * @param slippage                  Expected slippage from entering position in precise units (1% = 10^16)
+   * @param callerAddress             Address of the method caller
+   *
+   * @return                          Maximum amount of Sets that can be issued
+   */
+  public async getMaximumSetTokenIssueAmountAsync(
+    setTokenAddress: Address,
+    slippage: BigNumber,
+    callerAddress: Address = undefined,
+  ): Promise<BigNumber> {
+    this.assert.schema.isValidAddress('setTokenAddress', setTokenAddress);
+
+    return await this.perpV2LeverageModuleViewerWrapper.getMaximumSetTokenIssueAmount(
+      setTokenAddress,
+      slippage,
+      callerAddress,
+    );
+  }
+
+  /**
+   * Returns the position unit for total collateral value as defined by Perpetual Protocol.
+   *
+   * @param setTokenAddress           Instance of SetToken
+   * @param callerAddress             Address of the method caller
+   *
+   * @return                          Collateral token address
+   * @return                          Total collateral value position unit
+   */
+  public async getTotalCollateralUnitAsync(
+    setTokenAddress: Address,
+    callerAddress: Address = undefined,
+  ): Promise<[Address, BigNumber]> {
+    this.assert.schema.isValidAddress('setTokenAddress', setTokenAddress);
+
+    return await this.perpV2LeverageModuleViewerWrapper.getTotalCollateralUnit(
+      setTokenAddress,
+      callerAddress,
+    );
+  }
+
+  /**
+   * Returns relevant data for displaying current positions. Identifying info for each position plus current
+   * size, index price, and leverage of each vAsset with an open position is returned. The sum quantity of vUSDC
+   * is returned along with identifying info in last index of array.
+   *
+   * @param setTokenAddress           Instance of the SetToken
+   * @param callerAddress             Address of the method caller
+   *
+   * @return                          Array of info concerning size and leverage of current vAsset positions
```

Comment:
can we add more info here in comments when we have this more concrete?

Reply:
well i guess nvm you do have `VAssetDisplayInfo` in the return so i guess can figure out from looking there